goal_src/jak1/pc/pckernel-common.gc

```diff
@@ -902,15 +816,15 @@
       (format 0 "[PC] PC Settings found at '~S'...loading!~%" *pc-temp-string-1*)
       (unless (read-from-file obj *pc-temp-string-1*)
         (format 0 "[PC] PC Settings found at '~S' but could not be loaded, using defaults!~%" *pc-temp-string-1*)
-        (reset obj)))
+        (reset obj #t)))
     (format 0 "[PC] PC Settings not found at '~S'...initializing with defaults!~%" *pc-temp-string-1*))
   0)
 
 (defmethod new pc-settings ((allocation symbol) (type-to-make type))
   "make a new pc-settings"
   (let ((obj (object-new allocation type-to-make (the-as int (-> type-to-make size)))))
     ;; load defaults not covered by the file
-    (set-defaults! obj)
+    (reset obj #f)
```

Comment:
non-actionable, just a comment...

the diff is so large so I can't actually tell if this is a simplification or if it just makes the pc-settings more confusing with now an additional boolean....I tried to make things more explicit by separating concerns but I have a feeling that undoes some of that.

Reply:
I think a boolean that determines whether we actually call the OS functions or not is more convenient than duplicating a lot of the settings code. We had a method to reset to defaults, and other methods that just "set" the defaults... it didn't make much sense with how the rest of the system was designed, so I changed it for a boolean. It seems relatively straight-forward to me, at least.

Reply:
I agree with that, a lot of it has to be rewritten in my eyes because it requires too much intimate knowledge of the internals to avoid making a mistake -- I was trying to make small steps towards that and reduce the duplication over-time, but this just increases the complexity.